src/main/java/apoc/util/FileUtils.java

```diff
@@ -260,4 +260,10 @@ public static boolean inNeo4jOwnedDirectory(File f) throws IOException {
             "dbms.directories.tx_log",
             "unsupported.dbms.directories.neo4j_home"
     );
+
+    public static void closeReaderSafely(CountingReader reader) {
```

Comment:
couldn't have a try-with-resource done the same? just curious

Reply:
good question my initial implementation was with a `try-with-resource`:
https://github.com/conker84/neo4j-apoc-procedures/commit/e41b338778432ceb1f7c050044ff4e4a08ab2af7#diff-fc68dee7028a27c02a83473c5e03ca44R167

But a test did by @moxious returned this exception:
```
Neo.ClientError.Procedure.ProcedureCallFailed: Failed to invoke procedure `apoc.metrics.get`: Caused by: java.io.IOException: Stream closed
```

So in order to fix it, I removed the `try-with-resource` and closed it as the stream closes